lib/internal/Magento/Framework/Mview/View/Subscription.php

```diff
@@ -188,26 +198,32 @@ protected function getLinkedViews()
     }
 
     /**
-     * Build trigger statement for INSERT, UPDATE, DELETE events
+     * Returns prepared statement
      *
      * @param string $event
-     * @param \Magento\Framework\Mview\ViewInterface $view
+     * @param ViewInterface $view
      * @return string
      */
-    protected function buildStatement($event, $view)
+    private function getStatement(string $event, ViewInterface $view): string
     {
-        // Get the subscription for the specific view and specific table.
-        // We will use column name from it.
-        $subscriptions = $view->getSubscriptions() ?? [];
-        if (empty($subscriptions[$this->getTableName()])) {
-            return '';
-        }
-
-        $subscription = $subscriptions[$this->getTableName()];
-
-        // Get the changelog from View to get changelog column name.
-        $changelog = $view->getChangelog();
+        return $this->getSubscriptionStatement->execute(
+            $event,
+            $view,
+            $this->getTableName(),
+            $this->ignoredUpdateColumns
+        );
+    }
 
+    /**
+     * Build trigger statement for INSERT, UPDATE, DELETE events
+     *
+     * @param string $event
+     * @param ChangelogInterface $changelog
+     * @return string
+     * @deprecated see \Magento\Framework\Mview\View\GetSubscriptionStatement
```

Comment:
Could add some description why it was deprecated and use `@see \Magento\Framework\Mview\View\GetSubscriptionStatement` for reference?